Disaster-Response-Platform/mobile/DisasterResponsePlatform/app/src/main/java/com/example/disasterresponseplatform/ui/profile/ProfileFragment.kt

```diff
@@ -27,17 +43,186 @@ class ProfileFragment : Fragment() {
+        binding.apply {
+            // read image from url and set it to profilePhoto
+            if (user.profilePhoto != null) {
+                Picasso.get().load(user.profilePhoto!!.toUri()).into(profileImage)
+            }
+            profileFullname.text = user.name + " " + user.surname
+            profileUsername.text = user.username
+            profileEmail.text = user.email
+            profilePhoneNumber.text = user.phone
+            profileEmailVerifiedIcon.visibility = if (user.isEmailVerified) View.VISIBLE else View.GONE
+            profilePhoneVerifiedIcon.visibility = if (user.isPhoneVerified) View.VISIBLE else View.GONE
+            when (user) {
+                is CredibleUser -> {
+                    profileRegionLayout.visibility = View.VISIBLE
+                    profileRegion.text = "Credible User in " + user.region
+                }
+                is RoleBasedUser -> {
+                    profileProficiencyLayout.visibility = View.VISIBLE
+                    profileProficiency.text = "Profficient User: " + user.proficiency
+                }
+            }
+            when (user.verificationLevel) {
+                1 -> {
+                    profileVerifiedIcon.visibility = View.VISIBLE
+                    profileVerifiedByLayout.visibility = View.VISIBLE
+                    profileVerifiedBy.text = "Verified By " + user.verifiedBy
+                }
+                2 -> profileAdminIcon.visibility = View.VISIBLE
+            }
+
+            if (user.birth != null) {
+                profileBirthLayout.visibility = View.VISIBLE
+                profileBirth.text = user.birth.toString()
+            }
+
+            if (user.nationality != null) {
+                profileNationalityLayout.visibility = View.VISIBLE
+                profileNationality.text = user.nationality
+            }
+
+            if (user.idNumber != null) {
+                profileIdNumberLayout.visibility = View.VISIBLE
+                profileIdNumber.text = user.idNumber
+            }
+
+            if (user.address != null) {
+                profileAddressLayout.visibility = View.VISIBLE
+                profileAddress.text = user.address
+            }
+
+            if (user.education != null) {
+                profileEducationLayout.visibility = View.VISIBLE
+                profileEducation.text = user.education
+            }
+
+            if (user.healthCondition != null) {
+                profileHealthConditionLayout.visibility = View.VISIBLE
+                profileHealthCondition.text = user.healthCondition
+            }
+
+            if (user.bloodType != null) {
+                profileBloodTypeLayout.visibility = View.VISIBLE
+                profileBloodType.text = user.bloodType
+            }
+
+            var counter = 0;
+            for (socialMedia in user.socialMedia) {
+                val newView: View = LayoutInflater.from(requireContext()).inflate(R.layout.profile_item, null)
```

Comment:
You can use binding in here like:
val navViewBinding: ProfileItemBinding = ProfileItemBinding.inflate(LayoutInflater.from(requireContext()))
After that you can easily access binding objects like:
navViewBinding.profileItemText.text instead of  newView.findViewById<TextView>(R.id.profile_item_text).text